crates/floresta-chain/src/pruned_utreexo/chainparams.rs

```diff
@@ -50,6 +50,8 @@ pub struct ChainParams {
     /// A list of exceptions to the rules, where the key is the block hash and the value is the
     /// verification flags
     pub exceptions: HashMap<BlockHash, c_uint>,
+    /// Median Time Past (MTP). The MTP is calculated and defined on the run.
+    pub mtp: [u32; 11],
```

Comment:
I'm not sold on this, why not ask the API consumer to pass this? We could make a struct like `ValidationContext` or something with all this information and pass this to consensus. The consumer already has all this information, why not use it there?

Reply:
This is an option, but id prefer to have control of this one since the way i spent sometime thinking about it 
and fits on `floresta-chain` async design

Reply:
> it brings the best performance and fits on floresta-chain async design

Can you elaborate? I don't see that much difference in performance doing this or precomputing it using the headers.

Furthermore, block time verification should live in `accept_header`. We shouldn't waste bandwidth in a block with invalid timestamp, we download headers in a very non-linear way due to forks.